### ISSUE TITLE
Add zip to the RewriteCond in .htaccess.default

### ISSUE THREAD

**Issue by agoat** [contao/core#8321]
If the rewrite rule on line 263 and 264 in .htaccess(.default) ist set to rewrite URLs with no suffix
`RewriteCond %{REQUEST_FILENAME} !-d`
`RewriteRule .* index.php [L]`
then you can´t download any zip file from the files directory directly (like from the backend).

Thats because in the rewrite condition on line 262 the zip suffix is missing:
`  RewriteCond %{REQUEST_FILENAME} !\.(htm|php|js|css|map|htc|png|gif|jpe?g|ico|xml|csv|txt|swf|flv|mp4|webm|ogv|mp3|ogg|oga|eot|otf|tt[cf]|woff2?|svgz?|pdf|gz)$`

So we should add the zip suffix as well.

Tested on contao 3.5.12.

**Comment by fritzmg** [comment]
I cannot confirm that. Are you sure you set up your `.htaccess` correctly? The complete set of RewriteConds and -Rules should be
```
  RewriteCond %{REQUEST_FILENAME} !\.(htm|php|js|css|map|htc|png|gif|jpe?g|ico|xml|csv|txt|swf|flv|mp4|webm|ogv|mp3|ogg|oga|eot|otf|tt[cf]|woff2?|svgz?|pdf|gz)$
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteRule .* index.php [L]
```
Both `RewriteCond %{REQUEST_FILENAME} !-f` and `RewriteCond %{REQUEST_FILENAME} !-d` will prevent rewriting and URL to an actual file.

**Comment by agoat** [comment]
@fritzmg 
You are right! I´m not very familiar with the apache htaccess story. So I missed the fact that the `RewriteCond %{REQUEST_FILENAME} !-f` should stay and the ReriteRule have to be replaced by the extra RewriteCond and the RewriteRule for non suffixes.

**Comment by agoat** [comment]
I made some more test. 
If I try to download a file with the zip suffix that do not exist, I will get a no page error from contao, because the URL will be rewritten. When I try to download a file with e.g. a jpg suffix I will get an error from apache, that the file don´t exist.

So I think we should still adding the zip suffix to the RewriteCond to have the same behavior like jpgs or xml files.

**Comment by fritzmg** [comment]
With
```
RewriteCond %{REQUEST_FILENAME} !-d
```
requests like
```
http://www.example.org/files/foo.zip
```
should not be rewritten, if the folder `files` does exist (even if the file `foo.zip` does not exist).

**Comment by agoat** [comment]
I get a no page error with

`
www.example.org/files/foo.zip
`

**Comment by fritzmg** [comment]
> I get a no page error with

> www.example.org/files/foo.zip

Hm, you are right, I guess I must have done something wrong during testing. zip should be added to the list.

**Comment by agoat** [comment]
Your answered during I was fighting with the markdown styling ;-)

Ok. Then it´s all right.
I just started to wonder if I have a uncommon configuration on my server..

**Comment by agoat** [comment]
No comment??
Is there a reason to not add the zip suffix? gz and pdf is in the list as well. So why not adding zip?

I mean it´s only a tiny thing, but it is necessary if you want folder like urls.

**Comment by leofeyer** [comment]
I'll add it. But it is definitely not necessary to use folder URLs (as @fritzmg explained above).

**Comment by agoat** [comment]
I did not mean the folder URLs. It is necessary if you use no URL suffix `http://demo.contao.org/en/about-contao` (without the .html).

**Comment by leofeyer** [comment]
Even then it should not be necessary. The whole purpose of line 262 is to stop rewriting for known file types to reduce the number of IO lookups. You should be able to delete the line and your rewrite rules should still work!

**Comment by agoat** [comment]
Ok, I tested it on a clean installation and it´s like you wrote!!
I first thought it was up to another rewrite rule I had above:
```
RewriteCond %{HTTPS} !=on
RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI} [R=301,L]
```

So I tested it again on my production server and it all worked well. Don´t know why I had that problem.

But I think it´s not a fault to add zip to the RewriteCond. We allow it by default for upload, so we can exclude it here, too.
